indico/modules/rb/controllers/user/rooms.py

```diff
@@ -156,19 +158,42 @@ def _checkParams(self):
         self._room = Room.get(request.view_args['roomID'])
         self._occupancy_period = request.args.get('period', 'pastmonth')
         self._end = date.today()
+        self._year = None
+        self._month = None
         if self._occupancy_period == 'pastmonth':
             self._end = self._end - relativedelta(days=1)
             self._start = self._end - relativedelta(days=29)
         elif self._occupancy_period == 'thisyear':
             self._start = date(self._end.year, 1, 1)
         elif self._occupancy_period == 'sinceever':
-            self._start = Reservation.query.with_entities(func.min(Reservation.start_dt)).one()[0].date()
+            oldest = (db.session.query(Reservation.start_dt)
+                      .filter(Reservation.room_id == self._room.id)
+                      .order_by(asc(Reservation.start_dt)).first()[0])
+            self._start = oldest_booking_date = oldest.date() if oldest else self._end
         else:
-            raise IndicoError('Invalid period specified')
+            match = re.match(r'(\d+)(?:-(\d+))?', self._occupancy_period)
+            try:
+                self._year = int(match.group(1))
+                self._month = int(match.group(2)) if match.group(2) else None
+            except ValueError:
+                raise IndicoError('Invalid period specified')
+            if self._month:
+                self._start = date(self._year, self._month, 1)
+                self._end = self._start + relativedelta(months=1)
+                self._occupancy_period = '{:d}-{:02d}'.format(self._year, self._month)
```

Comment:
doesn't matter since we don't have time travel (yet?), but for consistency i'd use `{:04d}` for the year